packages/serverless/src/model/database/user-group.ts

```diff
@@ -44,28 +44,40 @@ export class ModelUserGroup extends ModelGeneral<OptionalId<TUserGroupRecord>> {
     return matchedRecord === 1;
   }
 
-  public async listGroupByUserName(
-    userName: string,
+  /** List all groups of a user in this database given a user query. */
+  public async listGroupByUserQuery(
+    // Only allow filter by indexed fields to prevent implicit performance
+    // issues
+    userQuery: Partial<
+      Pick<TUserRecord, 'userName' | 'publicKey' | 'email' | '_id'>
+    >,
     options?: FindOptions
   ): Promise<string[]> {
-    const modelGroup = new ModelGroup(this.databaseName);
-    const groupsList = await modelGroup.find(
+    const imUser = new ModelUser();
+    const user = await imUser.findOne(userQuery, options);
+    if (!user) {
+      throw new Error('User not found');
+    }
+
+    const imGroup = new ModelGroup(this.databaseName);
+    const groupsList = imGroup.find(
       {
-        _id: { $in: await this.listGroupId(userName) },
+        _id: { $in: await this.listGroupId(user.userName) },
       },
       options
     );
+
     return groupsList.map((group) => group.groupName).toArray();
   }
 
   public async listGroupId(userName: string): Promise<ObjectId[]> {
-    const userGroups = await this.find({ userName });
+    const userGroups = this.find({ userName });
     return userGroups.map((userGroup) => userGroup.groupOjectId).toArray();
   }
 
   public async groupNameToGroupId(groupName: string[]): Promise<ObjectId[]> {
     const modelGroup = new ModelGroup(this.databaseName);
-    const availableGroups = await modelGroup.find({
+    const availableGroups = modelGroup.find({
       groupName: { $in: groupName },
     });
```

Comment:
availableGroupList, can you double check the find that does find need await or not

Reply:
fixed